src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp

```diff
@@ -166,13 +170,20 @@ namespace ionization
                 /* kernel call for weighted ion density calculation */
                 density->template computeValue< CORE + BORDER, DensitySolver >(*srcSpecies, currentStep);
                 dc.releaseData( SrcSpecies::FrameType::getName() );
+                EventTask densityEvent = density->asyncCommunication( __getTransactionEvent() );
+                densityEvent += density->asyncCommunicationGather( densityEvent );
```

Comment:
Just for me again: the first of the two lines above gets the event of the `FieldTmp` operation and the second line says that in addition to this event the results from the previous event in the `BORDER` region of the neigboring GPU have ton be communicated to the `GUARD` of this GPU. Only afterwards the density values for particles whose shape extends outside the `BORDER` region will be correct, right?